src/transport/transport.h

```diff
@@ -224,33 +231,37 @@ class ConvectionTransport : public TransportBase {
      */
     EqData data_;
 
+    //@{
     /**
-     * Indicates if we finished the matrix and add vector by scaling with timestep factor.
+     * Flag indicates the state of object (transport matrix or source term).
+     * If false, the object is freshly assembled and not rescaled.
+     * If true, the object is scaled (not necessarily with the current time step).
      */
-	bool is_convection_matrix_scaled, need_time_rescaling;
-
-    double *sources_corr;
-    Vec v_sources_corr;
+	bool is_convection_matrix_scaled, is_src_term_scaled;
+    //@}
+    
+    double **sources_corr;
+    Vec *v_sources_corr;
     
-    ///temporary arrays to store constant values of fields over time interval
-    //(avoiding calling "field.value()" too often)
-    double **sources_density, 
-           **sources_conc,
-           **sources_sigma;
 
     TimeMark::Type target_mark_type;    ///< TimeMark type for time marks denoting end of every time interval where transport matrix remains constant.
-    double cfl_max_step;
-            // only local part
-
+    double cfl_max_step;    ///< Time step constraint coming from CFL condition.
+    
+    Vec vcfl_flow_,     ///< Parallel vector for flow contribution to CFL condition.
```

Comment:
This is due to the fact, that cfl is computed in two different parts of assembly - 'flow' and sources.
If e.g. only sources change, we update only vcfl_sources and use vcfl_flow from previous time step. If these vectors are only temporal, then I will need to recompute cfl from flow and sources every time any data or flow is changed. 

Cfl condition is evaluated while assembling matrix and sources. If we separate it, we  will do lots of computation again, I think.

Current approach is:
- (if data, or flow changed) assembly matrix and sources while the CFL is evaluated
- time step is selected
- matrix is scaled

From your other comments, I understand following approach:
- splitting create_transport matrix_mpi into smaller functions
- compute CFL independently, ahead of assembly (-> some data and fluxes would be evaluated twice)
- then use only temporal vectors for cfl computation (-> save memory, possibly increase time by recomputation of cfl terms)
- then in TOS we would evaluate CFL (without assembly), then choose timestep and then assembly matrix
- then it would be possible to assemble and scale the matrix at once